Tests/GRPCProtobufParserTests/ProtobufToCodeGenParserTests.swift

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GRPCCodeGen
+import GRPCProtobufParser
+import SwiftProtobuf
+import SwiftProtobufPluginLibrary
+import XCTest
+
+final class ProtobufToCodeGenParserTests: XCTestCase {
+  func testParser() throws {
+    let parsedCodeGenRequest = try ProtobufToCodeGenParser().parse(
+      input: self.helloWorldFileDescriptor
+    )
+    XCTAssertEqual(parsedCodeGenRequest.fileName, "helloworld.proto")
+    XCTAssertEqual(
+      parsedCodeGenRequest.leadingTrivia,
+      """
+      // DO NOT EDIT.
+      // swift-format-ignore-file
+      //
+      // Generated by the Swift generator plugin for the protocol buffer compiler.
+      // Source: helloworld.proto
+      //
+      // For information on using the generated types, please see the documentation:
+      //   https://github.com/grpc/grpc-swift
+
+      // Copyright 2015 gRPC authors.
+      //
+      // Licensed under the Apache License, \
+      Version 2.0 (the \"License\");
```

Comment:
No need to escape the newline, or the double quotes for that matter 

```suggestion
      // Licensed under the Apache License, Version 2.0 (the "License");
```